mllib/src/test/scala/org/apache/spark/ml/classification/ClassifierSuite.scala

```diff
@@ -29,4 +69,31 @@ object ClassifierSuite {
     "rawPredictionCol" -> "myRawPrediction"
   )
 
+  class MockClassifier(override val uid: String)
+    extends Classifier[Vector, MockClassifier, MockClassificationModel] {
+
+    def this() = this(Identifiable.randomUID("mockclassifier"))
+
+    override def copy(extra: ParamMap): MockClassifier = throw new NotImplementedError()
```

Comment:
Not sure; I've just followed the rest of the code base so far.  Probably just a tendency to avoid esoteric Scala notation.